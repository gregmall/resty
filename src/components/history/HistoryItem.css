section{
  display: flex;
  flex-wrap: wrap;
  padding: 5px;
  border-radius: 8px;
  background-color: rgb(238, 152, 95);
  margin: 5px;
}
