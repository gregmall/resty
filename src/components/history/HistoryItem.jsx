import React from 'react';
import PropTypes from 'prop-types';
import styles from './HistoryItem.css'
const HistoryItem = ({ url, method }) => (
  <section className={styles.HistoryItem}>
    <p>Method: {method}</p>
    <p>  Url: {url}</p>
  </section>
);

HistoryItem.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired
};

export default HistoryItem;
