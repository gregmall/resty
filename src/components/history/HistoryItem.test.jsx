import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryItem from './HistoryItem'

describe('HistoryItem component', () => {
  afterAll(() => cleanup());
  
  it('displays a history item', () => {
    const { asFragment } = render(<HistoryItem
      url="www.awebwsite.com"
      method="GET"
      />);
      expect(asFragment()).toMatchSnapshot();
  })
})
