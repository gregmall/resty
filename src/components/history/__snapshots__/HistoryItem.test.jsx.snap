// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem component displays a history item 1`] = `
<DocumentFragment>
  <section
    class="HistoryItem"
  >
    <p>
      Method: GET
    </p>
    <p>
        Url: www.awebwsite.com
    </p>
  </section>
</DocumentFragment>
`;
